src/overlays/actors/ovl_En_Pamera/z_en_pamera.c

```diff
@@ -22,135 +104,791 @@ const ActorInit En_Pamera_InitVars = {
+    }
+}
+
+void EnPamera_Update(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    EnPamera* this = THIS;
+
+    this->actionFunc(this, globalCtx);
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    func_80BD90AC(this, globalCtx);
+    func_80BD9384(this, globalCtx);
+    func_80BD94E0(this, globalCtx);
+    Collider_UpdateCylinder(&this->actor, &this->collider);
+    CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+    func_80BD93F4(this, globalCtx);
+}
+
+s32 EnPamera_OverrideLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot,
+                              Actor* thisx) {
+    EnPamera* this = THIS;
+
+    if (limbIndex == PAMERA_LIMB_HAIR) {
+        rot->x += this->limb9Rot.y;
+        rot->z += this->limb9Rot.x;
+    }
+    return 0;
+}
+
+void EnPamera_PostLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* thisx) {
+    if (limbIndex == PAMERA_LIMB_HAIR) {
+        SysMatrix_MultiplyVector3fByState(&D_80BDA5F0, &thisx->focus.pos);
+    }
+}
+
+void EnPamera_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    EnPamera* this = THIS;
+
+    OPEN_DISPS(globalCtx->state.gfxCtx);
+    func_8012C28C(globalCtx->state.gfxCtx);
+    gSPSegment(POLY_OPA_DISP++, 0x08, SEGMENTED_TO_VIRTUAL(D_80BDA604[this->unk_312]));
+    gSPSegment(POLY_OPA_DISP++, 0x09, SEGMENTED_TO_VIRTUAL(D_80BDA610[this->unk_314]));
+    gSPSegment(POLY_OPA_DISP++, 0x0A, SEGMENTED_TO_VIRTUAL(D_80BDA5FC[this->unk_310]));
+    SkelAnime_DrawSV(globalCtx, this->skelAnime.skeleton, this->skelAnime.limbDrawTbl, this->skelAnime.dListCount,
+                     EnPamera_OverrideLimbDraw, EnPamera_PostLimbDraw, &this->actor);
+    CLOSE_DISPS(globalCtx->state.gfxCtx);
+}
+
+void func_80BD9840(EnPamera* this, GlobalContext* globalCtx) {
+    this->actor.update = func_80BDA344;
+    this->actor.flags |= 0x2000000;
+    this->actor.flags |= 0x100000;
+    if ((gSaveContext.weekEventReg[75] & 0x20) || (gSaveContext.weekEventReg[52] & 0x20)) {
+        func_80BD9E60(this);
+        func_80BD9938(this);
+    } else {
+        this->actor.draw = NULL;
+        func_80BD9E60(this);
+        func_80BD9904(this);
+    }
+    if (gSaveContext.weekEventReg[14] & 4) {
+        func_801A0204(0x27);
+    } else {
+        func_801A0204(0x1F);
+    }
+}
+
+void func_80BD9904(EnPamera* this) {
+    this->actor.flags &= ~1;
+    this->actionFunc = &func_80BD9928;
+}
+
+void func_80BD9928(EnPamera* this, GlobalContext* globalCtx) {
+}
+
+void func_80BD9938(EnPamera* this) {
+    this->actionFunc = func_80BD994C;
+}
+
+void func_80BD994C(EnPamera* this, GlobalContext* globalCtx) {
+    if (func_800B84D0(&this->actor, globalCtx) != 0) {
+        if (Player_GetMask(globalCtx) == PLAYER_MASK_GIBDO_MASK) {
+            if (1) {
+                func_80BD93CC(this, 0, 1);
+                func_801518B0(globalCtx, 0x15A8, &this->actor);
+            }
```

Comment:
This is a weird one :o
Would this match too?
```suggestion
            if (1) { }
            func_80BD93CC(this, 0, 1);
            func_801518B0(globalCtx, 0x15A8, &this->actor);
```